consumer.go

```diff
@@ -213,6 +209,18 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 			}
 
 			shardIterator = resp.NextShardIterator
+			if c.immediateRescan && len(resp.Records) > 0 {
```

Comment:
if the `c.immediateRescan = true ` we'll skip the `<-ctx.Done()` below. wanna make sure the program will gracefully shutdown when told to

Reply:
Hm, wouldn't that get picked up by the `<-ctx.Done()` in the inner loop above?  Ok, there's a narrow window between the point where we finish iterating over all records (where `<-ctx.Done()` would apply) and looping back to `c.client.GetRecords()` but I see this as being very similar to the case of being asked to shutdown while having called out to `c.client.GetRecords()`: in either case, the Kinesis API call will complete quickly and we will fall through to the inner loop where `<-ctx.Done()` there would return.

Reply:
ahh yep, good point. didn't see the additional check in the loop above (collapsed in PR preview)